sh_scrapy/middlewares.py

```diff
@@ -60,6 +60,11 @@ def process_request(self, request, spider):
             request.meta[HS_PARENT_ID_KEY] = request_id
 
     def process_response(self, request, response, spider):
+        # This class of response check is intended to fix the bug described here
+        # https://github.com/scrapy-plugins/scrapy-zyte-api/issues/112
+        if type(response).__name__ == "DummyResponse":
```

Comment:
We just discussed this in a meeting with @kmike.
In addition to the name, can we also check the import path? Something like
```python
type(response).__module__ == "scrapy_poet.api"
```
or
```python
type(response).__module__.startswith("scrapy_poet")
```
if we don't have problems in case the import path changes.

It probably does not happen often, but I'm concerned that any user-defined `DummyResponse` will also trigger this code path.

Reply:
Thank you @elacuesta , the fix is added.